altapay.php

```diff
@@ -115,6 +115,33 @@ public function install()
         ) ENGINE=' . _MYSQL_ENGINE_ . '  DEFAULT CHARSET=utf8');
         }
 
+        Db::getInstance()->Execute('CREATE TABLE IF NOT EXISTS `' . _DB_PREFIX_ . 'altapay_order_reconciliation` (
+            `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+            `id_order` int(10) unsigned NOT NULL,
+            `reconciliation_identifier` varchar(255) NOT NULL,
+            `transaction_type` varchar(255) NOT NULL,            
+            PRIMARY KEY (`id`)
+        ) ENGINE=' . _MYSQL_ENGINE_ . '  DEFAULT CHARSET=utf8');
+
+        Db::getInstance()->Execute('INSERT INTO `' . _DB_PREFIX_ . 'request_sql` (`name`, `sql`) 
+            VALUES (\'AltaPay Order Reconciliation\', "SELECT SQL_CALC_FOUND_ROWS
+            a.`id_order` AS `ID`, `reference` AS `Reference`, `total_paid_tax_incl` AS `Total`, `payment` AS `Payment Method`, a.`date_add` AS `Dated`,
+            ao.unique_id AS `AltaPay Order ID`, ao.payment_id AS `Transaction ID`,
+            aor.reconciliation_identifier  AS `Reconciliation Identifier`, aor.transaction_type AS `Transaction Type`,
+            CONCAT(LEFT(c.`firstname`, 1), \'. \', c.`lastname`) AS `Customer`,
+            osl.`name` AS `Status`
+            FROM `' . _DB_PREFIX_ . 'orders` a
+            
+            LEFT JOIN `' . _DB_PREFIX_ . 'altapay_order` ao ON (ao.`id_order` = a.`id_order`)
+            LEFT JOIN `' . _DB_PREFIX_ . 'altapay_order_reconciliation` aor ON (aor.`id_order` = a.`id_order`)
+            LEFT JOIN `' . _DB_PREFIX_ . 'customer` c ON (c.`id_customer` = a.`id_customer`)
+            LEFT JOIN `' . _DB_PREFIX_ . 'order_state` os ON (os.`id_order_state` = a.`current_state`)
+            LEFT JOIN `' . _DB_PREFIX_ . 'order_state_lang` osl ON (os.`id_order_state` = osl.`id_order_state` AND osl.`id_lang` = 1)
+            WHERE 1
+            
+            ORDER BY a.`id_order` DESC;")');
```

Comment:
Does this scripts runs every time the plugin is installed / initialised?

How does the merchant that already has the plugin installed, but simply upgrades, gets this changes?

Reply:
The mentioned code resides in `install`  function which is only executed upon plugin installation. 
The plugin does not support upgrade. So, the merchant uninstalls and then installs the new version of the plugin. 
Since the plugin DB tables are left as is upon uninstallation, the `install` function code has checks to take care of the database schema if one already exists.